### PR TITLE
Android/45 share with us

### PR DESCRIPTION
**Pull request by kjanderson2**
Added share with us screen. It's not perfect and I'd love to add a spinner while it's capturing the image but I think it does everything it is supposed to!

### CODE REVIEW

File: Android/app/src/main/java/com/mehequanna/mmiw/MmiwActivity.kt [hunk 1]
```diff
@@ -117,6 +125,17 @@ class MmiwActivity : AppCompatActivity() {
         setViewOnClickerListeners()
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == SHARE_INTENT_CODE) {
+            if (resultCode == RESULT_OK) {
```
Comment (Mehequanna) [inline Android/app/src/main/java/com/mehequanna/mmiw/MmiwActivity.kt:131]: Could this be `if (resultCode == RESULT_OK || resultCode == RESULT_CANCLED)`?

File: Android/app/src/main/java/com/mehequanna/mmiw/MmiwActivity.kt [hunk 2]
```diff
@@ -255,6 +274,15 @@ class MmiwActivity : AppCompatActivity() {
     private fun pauseSceneView(pause: Boolean) =
         if (pause) sceneView.pause() else sceneView.resume()
 
+    // this sets all the views back to how they should be when capturing as if it were the first time
+    private fun resetToCapturing() {
```
Comment (Mehequanna) [inline Android/app/src/main/java/com/mehequanna/mmiw/MmiwActivity.kt:278]: 🔥 this is great!

File: Android/app/src/main/java/com/mehequanna/mmiw/MmiwActivity.kt [hunk 3]
```diff
@@ -383,8 +437,38 @@ class MmiwActivity : AppCompatActivity() {
...
+            view
+                .animate()
+                .alpha(1f)
+                .setDuration(200)
+                .setListener(null)
+                .start()
+        }
+    }
+
+    private fun fadeOutView(view: View?, onAnimationEnd: (() -> Unit)? = null) {
```
Comment (Mehequanna) [inline Android/app/src/main/java/com/mehequanna/mmiw/MmiwActivity.kt:453]: These are slick :)

**Review by Mehequanna — CHANGES_REQUESTED** [review]
This looks and works great! The only change I'm requesting is the share button is not red. Are you seeing that?


Screenshot_20210517-230759

**Comment by kjanderson2** [comment]
Weirdly it's having trouble being red and also having the ripple effect. I updated it ot be red but now it doesn't have the touch effect. I will keep noodling on that but at least we can get that in

File: Android/app/src/main/res/layout/share_with_us_layout.xml [hunk 4]
```diff
@@ -113,8 +113,10 @@
         android:layout_marginStart="16dp"
         android:layout_marginEnd="16dp"
         android:layout_marginBottom="8dp"
-        android:background="?attr/selectableItemBackground"
         android:backgroundTint="@color/red"
+        android:clickable="true"
+        android:focusable="true"
+        android:focusableInTouchMode="true"
```
Comment (Mehequanna) [inline Android/app/src/main/res/layout/share_with_us_layout.xml:119]: With this in here, I have to touch the button twice. I removed it and it seems to work with ripple. Is that the same for you?

**Review by Mehequanna — COMMENTED** [review]
One last thing, sorry :/

**Review by Mehequanna — APPROVED** [review]
This looks great! If we can get the share button to only be tapped once, feel free to merge :)

**Pull request merged**